pyvista/plotting/composite_mapper.py

```diff
@@ -0,0 +1,769 @@
+"""Module containing composite data mapper."""
+from itertools import cycle
+import logging
+import sys
+from typing import Optional
+import weakref
+
+import numpy as np
+
+from pyvista import _vtk
+from pyvista.utilities import convert_array, convert_string_array
+
+from ..utilities.misc import has_module, vtk_version_info
+from .colors import Color, get_cmap_safe
+
+
+class BlockAttributes:
+    """Block attributes used to set the attributes of a block.
+
+    Parameters
+    ----------
+    block : pyvista.DataObject
+        PyVista dataobject.
+
+    attr : pyvista.CompositeAttributes
+        Parent attributes.
+
+    Examples
+    --------
+    Add a sphere and a cube as a multiblock dataset to a plotter and then
+    change the visibility and color of the blocks.
+
+    >>> import pyvista as pv
+    >>> dataset = pv.MultiBlock([pv.Cube(), pv.Sphere(center=(0, 0, 1))])
+    >>> pl = pv.Plotter()
+    >>> actor, mapper = pl.add_composite(dataset)
+    >>> mapper.block_attr[1].color = 'b'
+    >>> mapper.block_attr[1].opacity = 0.1
+    >>> mapper.block_attr[1]
```

Comment:
If this 1-based API is here to stay we should definitely explain this too here in the class docs of `BlockAttributes`.

Reply:
Done.